newsfragments/3601.changed.md

```diff
@@ -0,0 +1 @@
+Deprecate `PyTryFrom` and `PyTryInto` traits.
```

Comment:
This should mention their replacement by `PyTypeCheck` and `PyTypeInfo`.

Reply:
Good idea, I also explicitly suggested `any.downcast()` as that's how users will typically engage with them.